json_serialization/reader.nim

```diff
@@ -178,6 +178,7 @@ func maxAbsValue(T: type[SomeInteger]): uint64 {.compileTime.} =
   elif T is int16: 32768'u64
   elif T is int32: 2147483648'u64
   elif T is int64: 9223372036854775808'u64
+  elif T is int: 9223372036854775808'u64
```

Comment:
that depends on `sizeof(int)`, i.e., i386 (32-bit cpu) uses `int32`.

wouldn't `high(T).uint64 + 1` work for `SomeSignedInt`? a bit weird that we have to hardcode these constants (even though this PR isn't the one introducing this methodology).

Reply:
Agree 100%, I'm also not sure why the values are hardcoded. I'll modify the function for something like what you suggested